File: graphview/src/main/java/de/blox/graphview/layered/SugiyamaConfiguration.java
package de.blox.graphview.layered;

/**
 *
 */
public class SugiyamaConfiguration {

    public static final int X_SEPARATION = 100;
    public static final int Y_SEPARATION = 100;

    private final int levelSeparation;
    private final int nodeSeparation;

    public int getLevelSeparation() {
        return levelSeparation;
    }

    public int getNodeSeparation() {
        return nodeSeparation;
    }

    public SugiyamaConfiguration(Builder builder) {
        levelSeparation = builder.levelSeparation;
        nodeSeparation = builder.nodeSeparation;
    }

    //TODO: enhance builder-pattern
    public static class Builder {

        private int levelSeparation = Y_SEPARATION;

        private int nodeSeparation = X_SEPARATION;

        public Builder setLevelSeparation(int levelSeparation) {
            this.levelSeparation = levelSeparation;
            return this;
        }

        public Builder setNodeSeparation(int nodeSeparation) {
            this.nodeSeparation = nodeSeparation;
            return this;
        }
    }
}

File: graphview/src/main/java/de/blox/graphview/layered/SugiyamaConfiguration.kt
package de.blox.graphview.layered

class SugiyamaConfiguration private constructor(builder: Builder) {
    val levelSeparation: Int = builder.levelSeparation
    val nodeSeparation: Int = builder.nodeSeparation

    class Builder {
        var levelSeparation = Y_SEPARATION
            private set
        var nodeSeparation = X_SEPARATION
            private set

        fun setLevelSeparation(levelSeparation: Int) = apply {
            this.levelSeparation = levelSeparation
        }

        fun setNodeSeparation(nodeSeparation: Int) = apply {
            this.nodeSeparation = nodeSeparation
        }

        fun build() = SugiyamaConfiguration(this)
    }

    companion object {
        const val X_SEPARATION = 100
        const val Y_SEPARATION = 100
    }
}
